lib/checkautovariables.cpp

```diff
@@ -309,8 +309,12 @@ bool CheckAutoVariables::checkAutoVariableAssignment(const Token *expr, bool inc
         }
         if (Token::simpleMatch(tok, "=")) {
             const Token *lhs = tok;
-            while (Token::Match(lhs->previous(), "%name%|.|*"))
-                lhs = lhs->previous();
+            while (Token::Match(lhs->previous(), "%name%|.|*|]")) {
+                if (lhs->link())
```

Comment:
should it be `if (lhs->previous()->link())` ?

Reply:
Yes, thanks.